app/src/main/java/com/android/bookswap/model/chat/FirebaseMessagingService.kt

```diff
@@ -0,0 +1,113 @@
+    val intent =
+        Intent(this, MainActivity::class.java).apply { addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP) }
+    val pendingIntent =
+        PendingIntent.getActivity(
+            this, 0, intent, PendingIntent.FLAG_ONE_SHOT or PendingIntent.FLAG_IMMUTABLE)
+
+    // Create intent and PendingIntent for the "Accept" action
+    val acceptIntent =
+        Intent(this, NotificationActionReceiver::class.java).apply { action = "ACTION_ACCEPT" }
+    val acceptPendingIntent =
+        PendingIntent.getBroadcast(
+            this,
+            0,
+            acceptIntent,
+            PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE)
+
+    // Create intent and PendingIntent for the "Decline" action
+    val declineIntent =
+        Intent(this, NotificationActionReceiver::class.java).apply { action = "ACTION_DECLINE" }
+    val declinePendingIntent =
+        PendingIntent.getBroadcast(
+            this,
+            0,
+            declineIntent,
+            PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE)
+
+    // Define a meaningful channel ID and channel name
+    val channelId = "bookswap_notifications_channel"
+    val channelName = "BookSwap Notifications"
+
+    // Build the notification
+    val notificationBuilder =
+        NotificationCompat.Builder(this, channelId)
+            .setSmallIcon(R.drawable.ic_launcher_foreground)
+            .setContentTitle(title)
+            .setContentText(messageBody)
+            .setAutoCancel(true)
+            .setContentIntent(pendingIntent)
+            .setPriority(NotificationCompat.PRIORITY_HIGH)
+            .addAction(R.drawable.ic_launcher_foreground, "Accept", acceptPendingIntent)
+            .addAction(R.drawable.ic_launcher_foreground, "Decline", declinePendingIntent)
+
+    // Get the NotificationManager
+    val notificationManager = getSystemService(NOTIFICATION_SERVICE) as NotificationManager
+
+    // Create the notification channel
+    createNotificationChannel(notificationManager, channelId, channelName)
+
+    // Use a unique ID for each notification to avoid overwriting
+    val notificationId = System.currentTimeMillis().toInt()
+    notificationManager.notify(notificationId, notificationBuilder.build())
+  }
+
+  /**
+   * Creates a notification channel for Android O and above.
+   *
+   * @param notificationManager the NotificationManager to create the channel.
+   * @param channelId the ID of the notification channel.
+   * @param channelName the name of the notification channel.
+   */
+  fun createNotificationChannel(
+      notificationManager: NotificationManager,
+      channelId: String,
+      channelName: String
+  ) {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+      val importance = NotificationManager.IMPORTANCE_HIGH
+      val channel =
+          NotificationChannel(channelId, channelName, importance).apply {
+            description = "Channel for BookSwap notifications"
```

Comment:
I don't know if this count as hard coded but if you think so, add also docstring